vxyowsup/tests/test_whatsapp.py

```diff
@@ -69,9 +74,12 @@ def assert_nodes_equal(self, node1, node2):
         self.assert_id_format_correct(node1)
         self.assert_id_format_correct(node2)
         id_stub = node1["id"].split('-')[0]
-        xml1 = node1.toString().replace(node1["id"], id_stub)
-        xml2 = node2.toString().replace(node2["id"], id_stub)
-        self.assertEqual(xml1, xml2)
+        self.assertEqual(node2["id"].split('-')[0], id_stub)
+        self.assertEqual(node1['to'], node2['to'])
+        self.assertEqual(node1.getChild('body'), node2.getChild('body'))
+        #xml1 = node1.toString().replace(node1["id"], id_stub)
+        #xml2 = node2.toString().replace(node2["id"], id_stub)
+        #self.assertEqual(xml1, xml2)
```

Comment:
The cause of the `[ENCODED DATA]` is one of the darker corners of Python 2. In Python 2 once can do `u"a" + "b"` and the result is roughly the same as `u"a" + b.decode(sys.getdefaultencoding())`. This means that if we pass any unicode to yowsup, it unexpectedly causes yowsup to internally convert some of its byte strings into unicode strings (which surprises yowsup and results in the `[ENCODED DATA]` string in the XML).